### ISSUE TITLE
[Feature] router adds add_worker_url, remove_worker_url api

### ISSUE THREAD

**Issue by 81549361** [sgl-project/sglang#2343]
### Checklist

- [X] 1. If the issue you raised is not a feature but a question, please raise a discussion at https://github.com/sgl-project/sglang/discussions/new/choose Otherwise, it will be closed.
- [X] 2. Please use English, otherwise it will be closed.

### Motivation

Thank you very much to the sglang team for open-sourcing such a useful inference framework.

We are currently deploying Llama3 70b on 3 X 8 X H100s. Run the command on each server separately:
```
python -m sglang_router.launch_server --model-path NousResearch/Hermes-3-Llama-3.1-70B-FP8 --port 6006 --dp-size 4 --tp 2 --router-eviction-interval 300 --context-length 20000
```
Then the three exposed http://127.0.0.1:6006 services are randomly routed by konga. Compared with directly using the --dp command, the cache rate is increased from 25% to 35%.

Why not use --worker-urls?

The current problem is that Rust routing currently supports cross-machine routing, but since workers cannot be added or deleted, when encountering occasional server errors or restarts, the router process needs to be restarted to reroute.

For our case (Long context, multi-round conversations ), if the routing is matched correctly, the cache rate should theoretically be able to reach 70-80%.

**Comment by Lzhang-hub** [comment]
+1

**Comment by ByronHsu** [comment]
Solid ask! I am aiming to implement a version by this week.

**Comment by ByronHsu** [comment]
@Lzhang-hub are you on slack? Let's chat about your use case as well. https://join.slack.com/t/sgl-fru7574/shared_invite/zt-2vmbanu9u-OwS5keCZHDaCcOGe6LTCPA

**Comment by Lzhang-hub** [comment]
yes, I'm already in slack, my use case is same as 81549361, also need a api to add or delete worker url.

**Comment by ByronHsu** [comment]
@Lzhang-hub what is your user name? can you ping me on slack? I am Byron Hsu

**Comment by Lzhang-hub** [comment]
ok
